Toolkit/ArcGISToolkit/AR/ArcGISARView.swift

```diff
@@ -0,0 +1,512 @@
+//
+// Copyright 2019 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ARKit
+import ArcGIS
+
+extension simd_quatd {
+    init(statusBarOrientation: UIDeviceOrientation) {
+        switch statusBarOrientation {
+        case .landscapeLeft:
+            self.init(ix: 0, iy: 0, iz: 0, r: 1)
+        case .landscapeRight:
+            self.init(ix: 0, iy: 0, iz: 1, r: 0)
+        case .portrait:
+            let squareRootOfOneHalf = (0.5 as Double).squareRoot()
+            self.init(ix: 0, iy: 0, iz: squareRootOfOneHalf, r: squareRootOfOneHalf)
+        case .portraitUpsideDown:
+            let squareRootOfOneHalf = (0.5 as Double).squareRoot()
+            self.init(ix: 0, iy: 0, iz: -squareRootOfOneHalf, r: squareRootOfOneHalf)
+        default:
+            // default to landscapeLeft
+            self.init(ix: 0, iy: 0, iz: 0, r: 1)
+        }
+    }
+}
+
+extension ArcGISARView.clError: CustomNSError {
+    static var errorDomain: String {
+        return kCLErrorDomain
+    }
+
+    var errorCode: Int {
+        switch self {
+        case .accessDenied:
+            return CLError.Code.denied.rawValue
+        case .missingPListKey:
+            return CLError.Code.denied.rawValue
+        }
+    }
+    
+    var errorDescription: String? {
+        switch self {
+        case .accessDenied:
+            return "Access to the device location is denied."
+        case .missingPListKey:
+            return "You must specify a location usage description key (NSLocationWhenInUseUsageDescription or NSLocationAlwaysUsageDescription) in your plist."
+        }
+    }
+}
+
+public class ArcGISARView: UIView {
+    
+    enum clError {
+        case accessDenied
+        case missingPListKey
+    }
+
+    // MARK: public properties
+    
+    /// The view used to display the `ARKit` camera image and 3D `SceneKit` content.
+    public lazy private(set) var arSCNView = ARSCNView(frame: .zero)
+    
+    /// The view used to display ArcGIS 3D content.
+    public lazy private(set) var sceneView = AGSSceneView(frame: .zero)
```

Comment:
I don't see a reason these should be lazy. They are used right away in sharedInitialization, which is called from the init. Couldn't it just be constants like so:

```swift
public let sceneView = AGSSceneView(frame: .zero)
```

Reply:
Done.  Made both `let`.